CMakeLists.txt

```diff
@@ -55,12 +55,16 @@ option(IREE_ENABLE_CLANG_TIDY "Builds IREE in with clang tidy enabled on IREE's
 option(IREE_ENABLE_CPUINFO "Enables runtime use of cpuinfo for processor topology detection." ON)
 
 option(IREE_BUILD_COMPILER "Builds the IREE compiler." ON)
-option(IREE_BUILD_TESTS "Builds IREE unit tests." ON)
 option(IREE_BUILD_DOCS "Builds IREE docs." OFF)
 option(IREE_BUILD_SAMPLES "Builds IREE sample projects." ON)
 option(IREE_BUILD_PYTHON_BINDINGS "Builds the IREE python bindings" OFF)
 option(IREE_BUILD_TRACY "Builds tracy server tools." OFF)
 
+option(IREE_BUILD_TESTS "Builds IREE unit tests." ON)
+# Tests under build_tools is less interesting to users and requires extra
+# dependencies. Disable them by default. Related issue: #10244
+cmake_dependent_option(IREE_ENABLE_BUILD_TOOL_TESTS "Enables tests of IREE build tools." OFF ${IREE_BUILD_TESTS} OFF)
+
```

Comment:
With the more test coverage in build_tools, the tests start pulling in some extra package dependencies. In this change we need the `requests` package.

I think it might make sense to disable the build tool tests by default, as the reason stated in the comments. #10244 could resolve this issue eventually.

Reply:
We already have `requests` in build_requirements.txt: https://cs.github.com/iree-org/iree/blob/ceb9cfd8c955f8fed500112702c7f7cafc2fc38d/runtime/bindings/python/iree/runtime/build_requirements.txt?q=build_requirements.txt. It's used for fetching Cuda stuff (maybe somewhere else as well). Honestly, it's such a ubiquitous package, that I don't think it's too bad to require it. Even the official Python docs recommend it over the standard library `urllib.request`.

Reply:
SG. I updated the website to mention the dependency on `requests`. There is a failing windows test will need #11696 to pass